teps/0061-allow-custom-task-to-be-embedded-in-pipeline.md

```diff
@@ -0,0 +1,239 @@
+2. Custom task specification verification/validation should be handed
+over to custom task controllers, custom task specification must not be
+validated by Tektoncd/Pipeline validation logic.
+3. Custom taskSpec should be submitted as part of the runSpec.
+
+### Non-Goals
+
+Custom task controllers are to be developed by other parties. Custom task
+specification should not be validated by Tektoncd/Pipeline webhooks.
+
+### Use Cases (optional)
+
+<!--
+Describe the concrete improvement specific groups of users will see if the
+Motivations in this doc result in a fix or feature.
+
+Consider both the user's role (are they a Task author? Catalog Task user?
+Cluster Admin? etc...) and experience (what workflows or actions are enhanced
+if this problem is solved?).
+-->
+
+- Users can put all the information in one pipelineRun CR.
+- Users don't have to manage the custom task CR. Since custom task CR is
+namespace scope. Currently if there are multiple users in the same namespace, they
+will have conflicts when they are using the same name for their custom task CR.
+- In KFP, we need all the templates and task spec live in each pipeline. Currently, 
+having all the custom task templates living in the Kubernetes namespace scope means that
+we have to make multiple API calls to Kubernetes in order to get all the pipeline
+information to render in our API/UI. 
+
+
+## Requirements
+
+<!--
+Describe constraints on the solution that must be met. Examples might include
+performance characteristics that must be met, specific edge cases that must
+be handled, or user scenarios that will be affected and must be accomodated.
+-->
+
+- The Tekton controller is responsible for adding the the custom task spec to
+the Run spec. Then it get validate by the custom task controllers.
+
+## Proposal
+TBD
+
+### Notes/Caveats (optional)
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate. Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the WGs or subproject.
+-->
+
+None
+
+### User Experience (optional)
+
+<!--
+Consideration about the user experience. Depending on the area of change,
+users may be task and pipeline editors, they may trigger task and pipeline
+runs or they may be responsible for monitoring the execution of runs,
+via CLI, dashboard or a monitoring system.
+
+Consider including folks that also work on CLI and dashboard.
+-->
+
+With the embedded taskSpec for the custom task, all the Tekton clients
+can create a pipeline or pipelineRun using a single API call to the Kubernetes.
+Any downstream systems that employ tektoncd e.g. Kubeflow pipelines, will not be
+ managing all the custom task CRs and their versioning.
+
+It is natural for a user to follow ways such as a 
```

Comment:
I believe you're referring here to the fact that deployments and such allow the user to define a podTemplate or similar, is that right?  Could you call that out specifically as part of this?  Otherwise it's left up to interpretation and prior knowledge/understanding of those resource types.

Reply:
will do